packages/webapp/src/routes/deployment/+page.svelte

```diff
@@ -0,0 +1,3174 @@
+using-words-from raindex-subparser
+epoch:call<'get-epoch>(),
+io: call<'io-for-epoch>(epoch),
+epoch-max-output: call<'amount-for-epoch>(epoch io),
+other-total-out
+_
+other-vwaio: call<'get-cost-basis-io-ratio>(),
+max-output: max(
+  epoch-max-output
+  every(
+    is-fast-exit
+    mul(other-total-out other-vwaio))),
+_: io,
+:call<'set-last-trade>(io);
+
+#handle-io
+min-trade-amount: mul(min-amount 0.9),
+:ensure(
+  greater-than-or-equal-to(
+    if(call<'amount-is-output>() output-vault-decrease() input-vault-increase())
+    min-trade-amount)
+  "Min trade amount."),
+:call<'set-cost-basis-io-ratio>();
+`;
+
+	let gui: DotrainOrderGui | undefined = undefined;
+	let availableDeployments: Record<string, { label: string }> = {};
+	onMount(async () => {
+		try {
+			let deployments: AvailableDeployments =
+				await DotrainOrderGui.getAvailableDeployments(dotrain);
+			availableDeployments = Object.fromEntries(
+				deployments.map((deployment) => [
+					deployment.deployment_name,
+					{
+						label: deployment.deployment_name,
+						deployment
+					}
+				])
+			);
+		} catch (error) {
+			// eslint-disable-next-line no-console
+			console.error('Failed to load deployments:', error);
+		}
+	});
+
+	let selectedDeployment: string | undefined = undefined;
+	async function handleDeploymentChange(deployment: string) {
+		if (!deployment) return;
+
+		try {
+			gui = await DotrainOrderGui.chooseDeployment(dotrain, deployment);
+		} catch (error) {
+			// eslint-disable-next-line no-console
+			console.error('Failed to get gui:', error);
+		}
+	}
+	$: if (selectedDeployment) {
+		handleDeploymentChange(selectedDeployment as string);
+	}
+
+	let tokenInfos: TokenInfos;
+	function getTokenInfos() {
+		if (!gui) return;
+		tokenInfos = gui.getTokenInfos();
+	}
+
+	let isFieldPreset: Record<string, boolean> = {};
+	let allFieldDefinitions: GuiFieldDefinition[] = [];
+	function getAllFieldDefinitions() {
+		if (!gui) return;
+		allFieldDefinitions = gui.getAllFieldDefinitions();
+		isFieldPreset = Object.fromEntries(allFieldDefinitions.map((field) => [field.binding, true]));
+	}
+
+	let isDepositPreset: Record<string, boolean> = {};
+	let allDeposits: GuiDeposit[] = [];
+	function getDeposits() {
+		if (!gui) return;
+		allDeposits = gui.getCurrentDeployment().deposits;
+		isDepositPreset = Object.fromEntries(allDeposits.map((deposit) => [deposit.token_name, true]));
+	}
+
+	$: if (gui) {
+		getTokenInfos();
+		getAllFieldDefinitions();
+		getDeposits();
+	}
+
+	async function handleAddOrder() {
+		try {
+			if (!gui) return;
+
+			// @ts-expect-error window.ethereum is not typed
+			const provider = new ethers.BrowserProvider(window.ethereum);
+			const signer = await provider.getSigner();
+			const address = await signer.getAddress();
+
+			// const allowances: AllowancesResult = await gui.checkAllowances(address);
+			// console.log(allowances);
+
+			const approvals: ApprovalCalldataResult = await gui.generateApprovalCalldatas(address);
```

Comment:
so this checks internally to see if the deposit amounts exceed the existing allowances onchain? also does this ignore deposits set to 0?